src/plugins/discover/public/application/components/sidebar/discover_field.tsx

```diff
@@ -311,6 +324,29 @@ export function DiscoverField({
             />
           </EuiFlexItem>
         )}
+        {canDeleteField && (
+          <EuiFlexItem grow={1} data-test-subj="discoverFieldListPanelDeleteItem">
+            <EuiToolTip
+              content={i18n.translate('discover.fieldChooser.discoverField.deleteFieldLabel', {
+                defaultMessage: 'Delete index pattern field',
+              })}
+            >
+              <EuiButtonIcon
+                onClick={() => {
+                  if (onDeleteField) {
+                    onDeleteField(field.name);
+                  }
+                }}
+                iconType="trash"
+                data-test-subj={`discoverFieldListPanelDelete-${field.name}`}
+                color="danger"
+                aria-label={i18n.translate('discover.fieldChooser.discoverField.deleteFieldLabel', {
+                  defaultMessage: 'Delete index pattern field',
```

Comment:
@flash1293 in Discover we refer to 'remove' as removing the field from the table, so I've decided to use `Delete` here. Think we could change this in Lens so well, so it's consistent?

Reply:
That makes a lot of sense, index pattern management is also using "delete" instead of "remove":
<img width="289" alt="Screenshot 2021-04-13 at 11 10 33" src="https://user-images.githubusercontent.com/1508364/114528072-f19b5780-9c48-11eb-8670-7c9898e744fc.png">